src/lightning/pytorch/strategies/launchers/multiprocessing.py

```diff
@@ -253,7 +252,9 @@ def update_main_process_results(self, trainer: "pl.Trainer", extra: Dict[str, An
         """
         # NOTE: `get_extra_results` needs to be called before
         callback_metrics = extra["callback_metrics"]
-        trainer.callback_metrics.update(apply_to_collection(callback_metrics, np.ndarray, lambda x: torch.tensor(x)))
+        trainer.callback_metrics.update(
+            apply_to_collection(callback_metrics, type(Tensor().numpy()), lambda x: torch.tensor(x))
```

Comment:
Is there a cleaner way to access numpy types (without `import numpy` obviously)?

Reply:
I suggest to remove this conversion altogether (L255). The counterpart in L239 should then just clone the tensor instead of calling numpy(). This would then cover the case that is described in the comment and a numpy conversion is no longer needed.

Reply:
@awaelchli I rebased and pushed a commit that does what I got from your comment. Let me know if this is what you meant.

Reply:
Hey @Peiffap thank you for trying my suggestion. The multiprocessing tests don't pass though, so my suggestion seems wrong. I haven't looked at it closely, so I don't know what's going on or what the solution could be. It might be good to remove the code changes from `multiprocessing.py` here and merge the PR, leaving this for a future investigation.

Reply:
Ah, I see. Thanks for giving it a shot at least. I've reverted the changes and I'll open an issue to see how to avoid casting to numpy without breaking things.

Reply:
perfect!